packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -184,7 +185,7 @@ export default declare((api, opts) => {
         }
 
         if (file.ast.comments) {
-          for (const comment of (file.ast.comments: Array<Object>)) {
+          for (const comment of file.ast.comments as Array<any>) {
```

Comment:
Guess we can remove the array annotation for the same rationale on https://github.com/babel/babel/pull/13221#discussion_r631435037.